test/functional/homepage_controller_test.rb

```diff
@@ -5,6 +5,68 @@ class HomepageControllerTest < ActionController::TestCase
     login_as_stub_user
   end
 
+  context "#confirm_destroy" do
+    should "render confirm destroy page for draft edition" do
+      popular_links = FactoryBot.create(:popular_links, state: "draft")
+
+      get :confirm_destroy, params: { id: popular_links.id }
+
+      assert_template "homepage/popular_links/confirm_destroy"
+    end
+
+    should "not render confirm destroy page if edition is published" do
+      popular_links = FactoryBot.create(:popular_links, state: "published")
+
+      get :confirm_destroy, params: { id: popular_links.id }
+
+      assert_redirected_to show_popular_links_path
+    end
+  end
+
+  context "destroy" do
+    setup do
+      @popular_links = FactoryBot.create(:popular_links, state: "draft")
+      get :confirm_destroy, params: { id: @popular_links.id }
+    end
+
+    should "delete the popular links from the database" do
```

Comment:
I think we can have another context block within destroy for the two states "state is draft" and "state is published". and then separate setup for both.